### PR TITLE
Elm 0.15 (new formula)

### PR DESCRIPTION
**Pull request by rtfeldman**
Adds a Formula for a binary distribution of the Elm programming language (version 0.15).

### CODE REVIEW

**Comment by DomT4** [comment]
Is there any way at all to [build this from source](http://elm-lang.org/Install.elm#build-from-source)?

File: Library/Formula/elm.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Elm < Formula
+  homepage "http://elm-lang.org"
+  url "https://github.com/kevva/elm-bin/raw/v1.4.1/vendor/osx/elm-platform-osx.tar.gz"
```
Comment (DomT4) [inline Library/Formula/elm.rb]: Is this an official link? I couldn't find reference to it anywhere on the website.

Comment (rtfeldman) [inline Library/Formula/elm.rb]: It's the official [`npm`](http://npmjs.com) repo: https://www.npmjs.com/package/elm

@evancz can presumably vouch for it, if need be :smiley:

**Comment by rtfeldman** [comment]
@DomT4 It can be [built from source](http://elm-lang.org/Install.elm#build-from-source), but that process relies on the Haskell build toolchain, which has been notoriously prone to breaking...that's why the NPM installer just uses these precompiled binaries instead.

File: Library/Formula/elm.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/elm.rb]: It looks okay, but a :+1: from @evancz is certainly welcome.

**Comment by DomT4** [comment]
I'll let @Homebrew/owners add their say on the binary/from-source issue here. We're usually fairly strict on requiring from-source, and we have the `install_cabal_package` method which I suspect would work here, but we've made exceptions before.

**Comment by mikemcquaid** [comment]
Needs to be from source or go into `homebrew/binary`

**Comment by rtfeldman** [comment]
@mikemcquaid `install_cabal_package` has `depends_on "ghc"` - but that will install the latest version of GHC, namely 7.10. Building from source requires GHC 7.8; is there some way to specify this via `depends_on`? (I couldn't find one.)

It's really important to us that end users be able to `brew install elm` rather than having to go through `homebrew/binary`, so I want to make this work one way or another. If there's no way to avoid building this from source, what would the recommended way be to intentionally require end users have an older version of GHC installed so that we can build form source?

Thanks!

**Comment by mikemcquaid** [comment]
@rtfeldman Sorry for the delay. We don't support building with older versions of GHC, unfortunately, so the Homebrew way of doing this is to build your own version of GHC in the formula itself (obviously not practical in this case). I think it'll have to be `homebrew/binary`. Can I ask what your objection is to being in that tap?

**Comment by rtfeldman** [comment]
Separately putting the binaries there is fine, but I've specifically heard people wanting to do `brew install elm` - which I think is reasonable, as that's by far the most common Homebrew UX people have come to expect. :smiley:

**Comment by mikemcquaid** [comment]
They could `brew install homebrew/binary/elm` in that case (or `brew install elm` will return that in a search).

**Comment by DomT4** [comment]
What's the status on this?

**Comment by DomT4** [comment]
If this requires an older `ghc` to build from source let's add it to `homebrew/binary` per Mike's suggestion to work around that. Taps are fairly easy for users to find and use, and don't require any more manual intervention than `brew install homebrew/binary/elm`.

**Pull request closed without merging**